src/io/config.cpp

```diff
@@ -353,6 +358,9 @@ void BoostingConfig::GetTreeLearnerType(const std::unordered_map<std::string, st
     std::transform(value.begin(), value.end(), value.begin(), Common::tolower);
     if (value == std::string("serial")) {
       tree_learner_type = "serial";
+    } else if (value == std::string("gpu")) {
+      tree_learner_type = "serial";
+      device_type = "gpu";
```

Comment:
It is not necessary, just another shortcut and intuitive way to enable GPU. If you don't like this I can remove it.